runner.py

```diff
@@ -0,0 +1,653 @@
+            test_config['path'] = self.path
+
+            # Combine 'local' and 'global' config
+            for key in config:
+                if key not in test_config and key != 'tests':
+                    test_config[key] = config[key]
+
+            test = self.parse_test_case(test_config)
+            if test:
+                self.parsed_tests.append(test)
+
+        return self.parsed_tests
+
+
+def remove_prefix(string, prefix):
+    if string.startswith(prefix):
+        return string[len(prefix):]
+
+    return string
+
+
+def search_for_tests(paths=None):
+    if not paths:
+        return list(PRTOS_TEST_DIR.rglob('test*.yaml'))
+
+    test_paths = []
+    for path in paths:
+        candidate = list(path.rglob('test*.yaml'))
+        if not candidate:
+            raise ValueError(f'Test {path} does not contain .yaml test configuration')
+        test_paths.extend(candidate)
+
+    return test_paths
+
+
+def set_logger(level=logging.INFO):
+    root = logging.getLogger()
+    root.setLevel(logging.DEBUG)
+
+    stream_handler = logging.StreamHandler(sys.stdout)
+    stream_handler.setLevel(level)
+    formatter = logging.Formatter('%(message)s')
+    stream_handler.setFormatter(formatter)
+    root.addHandler(stream_handler)
+
+
+def args_file(arg):
+    path = pathlib.Path(arg)
+    if not path.exists():
+        print(f"Path {path} does not exist")
+        sys.exit(1)
+
+    path = path.resolve()
+    return path
+
+
+def parse_args():
+    verbosity = {'debug': 1, 'info': 2, 'warning': 3, 'error': 4}
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("-T", "--target",
+                        action='append', choices=TARGETS,
+                        help="Filter targets on which test will be build and ran. "
+                             "By default uses environment variable TARGET. If variable "
+                             "is not set then runs tests on all aviable targets. "
+                             "Flag can be used multiple times.")
+
+    parser.add_argument("-t", "--test",
+                        default=[], action='append', type=args_file,
+                        help="Specify directory in which test are will be searched. "
+                             "If flag is not used then runner search for tests in "
+                             "phoenix-rtos-tests dictionary. Flag can be used multiple times.")
+
+    parser.add_argument("--no-build",
+                        default=False, action='store_true',
+                        help="Omitt building step.")
+
+    parser.add_argument("-v", "--verbose",
+                        default='info',
+                        choices=sum(verbosity.items(), ()),
+                        help="Specify verbosity level. By default uses level info (2).")
+
+    args = parser.parse_args()
+
+    if isinstance(args.verbose, str):
+        args.verbose = verbosity[args.verbose]
+    # Convert to logging module's level (10, 20, 30, 40)
+    args.verbose *= 10
```

Comment:
I don't understand why you use two variable types `str` and `int` in the `choices` argument. 
There can only be one type, default `str` or set by `type` argument.
Now there is:
```
python3 runner.py -v 1
usage: runner.py [-h] [-T {ia32-generic}] [-t TEST] [--no-build]
                 [-v {debug,1,info,2,warning,3,error,4}]
runner.py: error: argument -v/--verbose: invalid choice: '1' (choose from 'debug', 1, 'info', 2, 'warning', 3, 'error', 4)
```
My proposition:
```suggestion
    parser.add_argument("-v", "--verbose",
                        default='info',
                        choices=tuple(verbosity.keys()),
                        help="Specify verbosity level. By default uses level 'info'.")

    args = parser.parse_args()

    # Convert to logging module's level (10, 20, 30, 40)
    args.verbose = verbosity[args.verbose] * 10
```

Reply:
I agree.